request_body_test.go

```diff
@@ -78,3 +79,85 @@ func TestRequestBody(t *testing.T) {
 		})
 	})
 }
+
+func TestRequestBody_WaitForAllBackends(t *testing.T) {
+	launchBackend := func(addr string) {
+		server := &http.Server{Addr: addr, Handler: http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			w.Header().Add("Delta-Backend", r.Header.Get("Delta-Backend"))
+			if r.Body != nil {
+				io.Copy(w, r.Body)
+			}
+		})}
+		e := server.ListenAndServe()
+		if e != nil {
+			t.Error(e)
+		}
+	}
+
+	go launchBackend(":18086")
```

Comment:
Instead of hardcoding ports for tests, get a random free port on the host system. Here's an example code:
```go
func freePort() (int, error) {
	l, err := net.Listen("tcp", "localhost:0")
	if err != nil {
		return 0, err
	}
	defer l.Close()
	return l.Addr().(*net.TCPAddr).Port, nil
}
```